printing_auto_base/models/printing_auto.py

```diff
@@ -0,0 +1,148 @@
+            ("report", "Report"),
+            ("attachment", "Attachment"),
+        ],
+        string="Data source",
+        default="report",
+        required=True,
+        help=(
+            "Choose to print the result of an odoo report or a pre-existing "
+            "attachment (useful for labels received from carriers that are "
+            "recorded on the picking as an attachment)"
+        ),
+    )
+    report_id = fields.Many2one("ir.actions.report")
+    attachment_domain = fields.Char("Attachment domain", default="[]")
+
+    condition = fields.Char(
+        "Condition",
+        default="[]",
+        help="Give a domain that must be valid for printing this",
+    )
+    record_change = fields.Char(
+        "Record change", help="Select on which document the report must be executed"
+    )
+
+    printer_id = fields.Many2one("printing.printer", "Printer")
+    printer_tray_id = fields.Many2one("printing.tray", "Tray")
+    nbr_of_copies = fields.Integer("Number of Copies", default=1)
+    action_on_error = fields.Selection(
+        [("log", "Record an error"), ("raise", "Raise an Exception")],
+        "Action on error",
+        default="log",
+        required=True,
+    )
+
+    @api.constrains("data_source", "report_id", "attachment_domain")
+    def _check_data_source(self):
+        for rec in self:
+            if rec.data_source == "report" and not rec.report_id:
+                raise UserError(_("Report is not set"))
+            if rec.data_source == "attachment" and (
+                not rec.attachment_domain or rec.attachment_domain == "[]"
+            ):
+                raise UserError(_("Attachment domain is not set"))
+
+    def _get_behaviour(self):
+        if self.printer_id:
+            result = {"printer": self.printer_id}
+            if self.printer_tray_id:
+                result["tray"] = self.printer_tray_id.system_name
+            return result
+        if self.data_source == "report":
+            return self.report_id.behaviour()
+        return self.env["ir.actions.report"]._get_user_default_print_behaviour()
+
+    def _get_record(self, record):
+        if self.record_change:
+            try:
+                return safe_eval(f"obj.{self.record_change}", {"obj": record})
+            except Exception as e:
+                raise ValidationError(
+                    _("The Record change could not be applied because: %s") % str(e)
+                ) from e
+        return record
+
+    def _check_condition(self, record):
+        domain = safe_eval(self.condition, {"env": self.env})
+        return record.filtered_domain(domain)
+
+    def _get_content(self, record):
+        generate_data_func = getattr(
+            self, f"_generate_data_from_{self.data_source}", None
+        )
+        content = []
+        if generate_data_func:
+            records = self._get_record(record)
+            for record in records:
+                content.append(generate_data_func(record)[0])
```

Comment:
Every result should be added to the content list. :-)
```suggestion
                content += generate_data_func(record)
```

Reply:
Fixed